analyzer/analyzer.go

```diff
@@ -42,23 +43,37 @@ func run(pass *analysis.Pass) (interface{}, error) {
 		if !ok {
 			return
 		}
-		if pass.TypesInfo.ObjectOf(called.Sel) != fmtSprintfObj {
-			return
-		}
-		if len(call.Args) != 2 {
-			return
-		}
+		calledObj := pass.TypesInfo.ObjectOf(called.Sel)
 
-		fmtString, value := call.Args[0], call.Args[1]
+		var (
+			fn    string
+			verb  string
+			value ast.Expr
+			err   error
+		)
+		switch {
+		case calledObj == fmtSprintObj && len(call.Args) == 1:
```

Comment:
Nice, where did you see some usages of this pattern ?
And where did you see that this could be faster ?

Reply:
What exactly "pattern" do you mean?
And what "exactly" could be faster?

Are you about analyzer code or about `Sprint`? 🙂

Reply:
About `fmt.Sprint`, sorry for the ambiguity

Reply:
Yes, sometimes people use `fmt.Sprint` for conversion or for strings join:

```go
fmt.Sprint(false) == fmt.Sprintf("%t", false)
fmt.Sprint(42) == fmt.Sprintf("%d", 42)
fmt.Sprint("a", "b") == strings.Join([]string{fmt.Sprintf("%s", "a"), fmt.Sprintf("%s", "b")}, "")
fmt.Sprint(10, 20) == strings.Join([]string{fmt.Sprintf("%d", 10), fmt.Sprintf("%d", 20)}, " ")
```

Or we have a typo after some refactoring:
```
fmt.Sprint("hello") == "hello"
```

Like probably here:
https://github.com/kubernetes/kubernetes/blob/02b0b79eda5a3d41a07a6f2965ee97d492e8b6b2/test/integration/daemonset/daemonset_test.go#L462

(it's a reason of https://staticcheck.io/docs/checks#S1039)

For me, this code looks strange and obviously this is low-performant:
```
$ make bench
BenchmarkStringFormatting/fmt.Sprint-8          16013566               116.7 ns/op
BenchmarkStringFormatting/fmt.Sprintf-8         17708608                59.19 ns/op
BenchmarkStringFormatting/just_string-8         1000000000               0.5123 ns/op

BenchmarkErrorFormatting/fmt.Sprint-8           12413080                94.45 ns/op
BenchmarkErrorFormatting/fmt.Sprintf-8          12557499                95.53 ns/op
BenchmarkErrorFormatting/Error()-8              339965853                3.449 ns/op

BenchmarkBoolFormatting/fmt.Sprint-8            11148936               104.4 ns/op
BenchmarkBoolFormatting/fmt.Sprintf-8           18491779                74.01 ns/op
BenchmarkBoolFormatting/strconv.FormatBool-8    1000000000               0.5458 ns/op

BenchmarkHexEncoding/fmt.Sprintf-8               6342842               176.9 ns/op
BenchmarkHexEncoding/hex.EncodeToString-8       39270228                35.31 ns/op

BenchmarkIntFormatting/fmt.Sprint-8              9539395               155.7 ns/op
BenchmarkIntFormatting/fmt.Sprintf-8             8850522               115.1 ns/op
BenchmarkIntFormatting/strconv.Itoa-8           23064060                49.10 ns/op

BenchmarkUintFormatting/fmt.Sprint-8            10711364               112.6 ns/op
BenchmarkUintFormatting/fmt.Sprintf-8           11290402               124.4 ns/op
BenchmarkUintFormatting/strconv.FormatUint-8    17183432                58.37 ns/op
```

Reply:
Thanks